test/block_labels_test.dart

```diff
@@ -0,0 +1,393 @@
+  Logic a;
+  Logic b;
+  String caseLabel;
+
+  LabeledCasesModule(
+      this.control, this.a, this.b, [this.caseLabel='cases']) {
+    control = addInput('control', control);
+    a = addInput('a', a);
+    b = addInput('b', b);
+    final out = addOutput('out');
+    out <= cases(
+      control, {LogicValue.zero: a, LogicValue.one: b},
+      label: caseLabel);
+  }
+}
+
+class LabeledCaseModule extends Module {
+  Logic a;
+  Logic b;
+  String firstCaseItemLabel;
+  String secondCaseItemLabel;
+  String defaultLabel;
+
+  LabeledCaseModule(
+      this.a, this.b, [this.firstCaseItemLabel='caseItem1',
+      this.secondCaseItemLabel='caseItem2', this.defaultLabel='default1']) {
+    a = addInput('a', a);
+    b = addInput('b', b);
+    final out = addOutput('out');
+
+    final aXorB = a ^ b;  
+    Combinational([
+      Case(
+        aXorB,
+        [
+          CaseItem(Const(LogicValue.ofString('0')),
+            [out < 1], label: firstCaseItemLabel),
+          CaseItem(Const(LogicValue.ofString('1')),
+            [out < 0], label: secondCaseItemLabel)
+        ], defaultItem: [out < 0], defaultLabel: defaultLabel),
+    ]);
+  }
+}
+
+class LabeledIfModule extends Module {
+  Logic a;
+  Logic b;
+
+  Logic get out => output('out');
+
+  String alwaysCombLabel;
+  String ifLabel;
+  String ifElseLabel;
+  String elseLabel;
+
+  LabeledIfModule(
+      this.a, this.b, [this.alwaysCombLabel='comb_1', this.ifLabel='if_1',
+      this.ifElseLabel='if_else_1', this.elseLabel='else_1']) {
+    a = addInput('a', a);
+    b = addInput('b', b);
+    final out = addOutput('out');
+
+    Combinational([
+      If.block([
+        Iff(a.eq(0) & b.eq(0), [
+          out < 0,
+        ], label: ifLabel),
+        ElseIf(a.eq(1) & b.eq(0), [
+          out < 1,
+        ], label: ifElseLabel),
+        Else([
+          out < 0,
+        ], label: elseLabel)
+      ]),
+    ], label: alwaysCombLabel);
+  }
+}
+
+class LabeledChainSequentialModule extends Module {
+  Logic reset;
+  Logic d;
+  Logic get q => output('q');
+  final clk = SimpleClockGenerator(10).clk;
+
+  String firstFfLabel;
+  String secondFfLabel;
+
+  LabeledChainSequentialModule(
+      this.reset, this.d, [this.firstFfLabel='ff_1',
+      this.secondFfLabel='ff_2']) {
+    reset = addInput('reset', reset);
+    d = addInput('d', d);
+    final q = addOutput('q');
+
+    final qInternal = LogicNet();
+    Sequential(
+      clk,
+      reset: reset,
+      resetValues: {
+        qInternal: 0
+      },
+      [
+        qInternal < d
+      ],
+      label: firstFfLabel
+    );
+    Sequential(
+      clk,
+      reset: reset,
+      resetValues: {
+        q: 0
+      },
+      [
+        q < qInternal
+      ],
+      label: secondFfLabel
+    );
+  }
+}
+
+class LabeledMultiBlockModule extends Module {
+  Logic a;
+  Logic b;
+  Logic c;
+  Logic d;
+
+  Logic get out_0 => output('out_0');
+  Logic get out_1 => output('out_1');
+
+  String firstBlockLabel;
```

Comment:
we should test that when *no* label is provided, the output SV looks right.  could make these `String?` (nullable) and have some tests which set them to `null`, then confirm compilation and correct verilog generation (you could even loop through and run all your tests with both a string and null provided, if you want)